src/CMakeLists.txt

```diff
@@ -74,7 +74,7 @@ endif(CMAKE_CXX_COMPILER_ID STREQUAL GNU)
 set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} ${CMAKE_CXX_FLAGS}")
 
 if(NOT CMAKE_BUILD_TYPE)
-  set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING "Default BUILD_TYPE is RelWithDebInfo, other options are: Debug, Release, and MinSizeRel." FORCE)
+  set(CMAKE_BUILD_TYPE "Debug" CACHE STRING "Default BUILD_TYPE is Debug, other options are: RelWithDebInfo, Release, and MinSizeRel." FORCE)
```

Comment:
this changes how the rpmbuild builds ceph packages. before this change, it builds with `CMAKE_BUILD_TYPE=RelWithDebInfo`, after this change it builds with `CMAKE_BUILD_TYPE=Debug`, which is not expected.

unlike debian/rules where we use the default parametersdh helper specified by `dh_auto_configure`, which passes `-DCMAKE_BUILD_TYPE=None` explicitly, `ceph.spec.in` does not pass this setting to cmake. i also checked "cmake-rpm-macros" package for a reference, neither does set `CMAKE_BUILD_TYPE`. see https://github.com/open-io/rpm-specfiles/blob/master/cmake/macros.cmake for a (probably outdated) copy of the macros defined by it. BTW, we are not using any of these macros, as this package is not available in RHEL / CentOS 7.

so i think a better fix is
```cmake

if(NOT CMAKE_BUILD_TYPE)
  if(EXISTS "${CMAKE_SOURCE_DIR}/.git")
    set(default_build_type "Debug")
  else()
    set(default_build_type "RelWithDebInfo")
  endif()
  set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE
      STRING "Default BUILD_TYPE is Debug, other options are: RelWithDebInfo, Release, and MinSizeRel." FORCE)
endif()
```

Reply:
Yes, this looks much better. Thanks a lot @tchaikov